### ISSUE TITLE
Fields in Neo4j Types of LocalTime and Time are being written as individual fields instead of single Struct Field

### ISSUE THREAD

**Issue by hshah-infa** [neo4j-contrib/neo4j-spark-connector#444]
## Guidelines

Please note that GitHub issues are only meant for bug reports/feature requests. If you have questions on how to use the Neo4j Connector for Apache Spark,
please ask on [the Neo4j Discussion Forum](https://community.neo4j.com/c/integrations/18) instead of creating an issue here.

## Expected Behavior (Mandatory)
When dataframe contains Temporal types such as LocalTime and Time, they are expected to be written as single Struct Field.
For example, dataframe containing column with name timeType which is of Neo4J Time type, contains 2 fields: type and value.
When timeType  is to be written to Neo4J, it is expected to be written as single field, let's say timeTypeTarget.

## Actual Behavior (Mandatory)
timeType field is being written as 2 fields in Neo4J: timeType.type and timeType.value.
Issue is seen for Neo4j Types: Time and LocalTime

## How to Reproduce the Problem
Standalone program, using Neo4j Spark Connector, reading from node containing Time or LocalTime and writing another node in Neo4J reproduces the issue.

Issue is within the Neo4JUtil.scala file where Value subfield is being read as Java String instead of UTF8String.
Type and Value subfields are both stored as UTF8String in the Spark dataframe by default.
Type field is being read correctly as UTF8String.
Value is incorrectly read as Java String. 

ClassCastException is thrown here: https://github.com/neo4j-contrib/neo4j-spark-connector/blob/a8e59efaf459bd76082fe9d7e78c4aff19b4308f/common/src/main/scala/org/neo4j/spark/util/Neo4jUtil.scala#L187

And then it is handled here: https://github.com/neo4j-contrib/neo4j-spark-connector/blob/a8e59efaf459bd76082fe9d7e78c4aff19b4308f/common/src/main/scala/org/neo4j/spark/util/Neo4jUtil.scala#L192

Which results the type being read as Map instead of Neo4J type.

### Simple Dataset (where it's possible)

```
// Insert the output of the `df.show()` call

+------------------------------------------------------------------------+------------------------+-----------+---------------+--------------+-----------------------+------------------------------------+-----------------------+----------+
|title                                                                   |**time.value**              |**time.type**  |**localTime.value**|**localTime.type**|localDateTime          |duration                            |dateTime               |date      |
+------------------------------------------------------------------------+------------------------+-----------+---------------+--------------+-----------------------+------------------------------------+-----------------------+----------+
|Cypher Basics I                                                         |12:50:35.556+01:00      |offset-time|12:50:35.556   |local-time    |2015-07-04 12:32:24    |{duration, 0, 0, 135, 0, P0M0DT135S}|2019-06-01 10:40:32.142|2019-06-01|
|Cypher Basics II                                                        |12:31:14.645876123-01:00|offset-time|12:31:14.645876|local-time    |1984-11-11 04:31:14    |{duration, 0, 0, 150, 0, P0M0DT150S}|2019-06-02 02:23:32.122|2019-06-02|
|Dates, Datetimes, Durations, LocalDateTime, LocalTime and Ttime in Neo4j|02:58:03.389Z           |offset-time|02:58:03.389   |local-time    |2022-09-06 19:58:03.389|{duration, 0, 0, 210, 0, P0M0DT210S}|2022-09-06 19:58:03.389|2022-09-07|
|Cypher Basics I                                                         |12:50:35.556+01:00      |offset-time|12:50:35.556   |local-time    |2015-07-04 12:32:24    |{duration, 0, 0, 135, 0, P0M0DT135S}|2019-06-01 10:40:32.142|2019-06-01|
|Cypher Basics II                                                        |12:31:14.645876123-01:00|offset-time|12:31:14.645876|local-time    |1984-11-11 04:31:14    |{duration, 0, 0, 150, 0, P0M0DT150S}|2019-06-02 02:23:32.122|2019-06-02|
|Dates, Datetimes, Durations, LocalDateTime, LocalTime and Ttime in Neo4j|02:58:03.389Z           |offset-time|02:58:03.389   |local-time    |2022-09-06 19:58:03.389|{duration, 0, 0, 210, 0, P0M0DT210S}|2022-09-06 19:58:03.389|2022-09-07|
+------------------------------------------------------------------------+------------------------+-----------+---------------+--------------+-----------------------+------------------------------------+-----------------------+----------+



### Steps (Mandatory)

1. Create a Node in Neo4J containing LocalTime and Time Neo4j Types
2. Create a standalone program consuming Neo4J Spark Connector. (Spark version being used is Spark 3)
3. Read the node as dataframe and write the dataframe creating another node.
4. Resulting node would contain the localTime and Time fields as individual fields

## Screenshots (where it's possible)

![image](https://user-images.githubusercontent.com/85647400/188783795-b9fb89c3-fce9-4ee9-8963-955e74dbf800.png)


![image](https://user-images.githubusercontent.com/85647400/188783735-05fb562b-2456-41d2-a4a3-b0be0f2d2c44.png)


## Specifications (Mandatory)

Currently used versions:
- Spark: 3.2.0
- Scala: 2.12.15
- Neo4j: 4.4.5
- Neo4j Connector: neo4j-connector-apache-spark_2.12_4.1.0_for_spark_3


### Versions

- Spark: 3.2.0
- Scala: 2.12.15
- Neo4j: 4.4.5
- Neo4j Connector: neo4j-connector-apache-spark_2.12_4.1.0_for_spark_3

## Additional information

* The code of the Spark job
* the structure of the Dataframe
* did you define the constraints/indexes?
* if you're you using any Spark Cloud provider please specify it (ie: Databricks)

**Comment by hshah-infa** [comment]
Created a pull request: https://github.com/neo4j-contrib/neo4j-spark-connector/pull/450

**Comment by conker84** [comment]
closed via [48dfa39](https://github.com/neo4j-contrib/neo4j-spark-connector/commit/48dfa393899b8b7af8595aa3e4fd598d90a33366)
